### PR TITLE
Use bazel to run gazelle and kazel by default

### PR DESCRIPTION
**Pull request by fejta**
/assign @ixdy @BenTheElder 

If `bazel` is installed then `bazel run //:gazelle`. Otherwise do the `go install gazelle && gazelle` like we did previously (after printing a warning message).

### CODE REVIEW

File: hack/update-bazel.sh [hunk 1]
```diff
@@ -17,19 +17,28 @@ set -o errexit
...
+touch "./vendor/BUILD.bazel"
 
-"${OUTPUT_GOBIN}/kazel"
+if ! which bazel > /dev/null; then
+  echo "Bazel is the preferred way to build and test the test-infra repo." >&2
+  echo "Please install bazel at https://bazel.build/ (future commits may require it)" >&2
+  deprecated-update
+  exit 0
+fi
+bazel run //:gazelle
```
Comment (ixdy) [inline hack/update-bazel.sh]: are we sure that the bazel version of gazelle matches the version we have vendored?

Comment (fejta) [reply hack/update-bazel.sh]: Versions can differ.

My thinking is:
a) wait for this to demonstrate itself as a problem before choosing a solution.
b) Remove the `deprecated-update` path before then.

File: hack/update-bazel.sh [hunk 2]
```diff
@@ -17,19 +17,28 @@ set -o errexit
...
-  -external=vendored \
-  -mode=fix
+# Ensure ./vendor/BUILD.bazel exists
+mkdir -p ./vendor
+touch "./vendor/BUILD.bazel"
 
-"${OUTPUT_GOBIN}/kazel"
+if ! which bazel > /dev/null; then
+  echo "Bazel is the preferred way to build and test the test-infra repo." >&2
+  echo "Please install bazel at https://bazel.build/ (future commits may require it)" >&2
```
[the remark this answers is not in the document]
Comment (fejta) [reply hack/update-bazel.sh]: Isn't the main landing page, which has a giant "Get Bazel" button that correctly detects your platform at least as useful?

[the remark this answers is not in the document]
Comment (BenTheElder) [reply hack/update-bazel.sh]: /shrug
people are bad at clicking links, less links for the lazy :^)

File: hack/update-bazel.sh (hunk 1 above)
Comment (BenTheElder) [reply hack/update-bazel.sh]: this_is_fine

**Pull request merged**